master/buildbot/db/logs.py

```diff
@@ -328,7 +328,8 @@ def thdcompressLog(conn):
             # calculate how many bytes we saved
             q = sa.select([sa.func.sum(sa.func.length(tbl.c.content))])
             q = q.where(tbl.c.logid == logid)
-            newsize = conn.execute(q).fetchone()[0]
+            # if fetchone()[0] is None, then set it to 0
+            newsize = conn.execute(q).fetchone()[0] or 0
```

Comment:
Normally, there won't be a reason for it to return None.

I have fixed the reason the code puked.  This change still makes the code more fault tolerant, but does not hide the real cause.  It just prevents a secondary crash which could be misleading in determining the cause.

In our case for buildbot-8 we had to subclass the LogLineObserver to relay important status and other information about the step since the output was not available until completion of the step.
For buildbot-9 we get stdout near realtime, so they were not needed.  I removed all those (but missed one), except the one that retrieved a url link  to a custom test system status.  Since the RealTimeLogLineObserver class had been simplified, the one use case I missed caused it to traceback.  That traceback triggered the fetchone()[0] to return None, which caused the log system to also crash.

This change just makes the logging code more robust.  It will not affect normal operation.  It will only prevent the log system from crashing due to some other crash like the one which was caused by my missed removal of the RealtimeLogLineObserver use in the build script.